NMS-14825: Remove UI Preview references in docs (#5447)
OpenNMS/opennms: docs/modules/operation/pages/deep-dive/user-management/security-roles.adoc

[[ga-role-user-management-roles]]
= Assign User Permissions

You can control user permission levels by creating and assigning security roles.
These roles regulate access to the {page-component-title} web UI and the REST API to exchange monitoring and inventory information.
ifndef::opennms-prime[]

In a distributed installation, Minion instances require an account that is assigned the `ROLE_MINION` security role to interact with {page-component-title}.
You can create one account for all Minions to share, or one account per Minion.
endif::opennms-prime[]

== Built-in security roles

The following security roles are built in to {page-component-title} by default.
Roles marked with an asterisk are the most commonly used.

[options="header", "autowidth"]
[cols="1,3"]
|===
| Security Role Name
| Description

> | ROLE_ADMIN \*

Reviewer: ```suggestion
| ROLE_ADMIN ^*^
```
The other syntax was showing the backslash. Carat worked locally for me.

| Permissions to create, read, update, and delete in the web UI and the REST API (see `ROLE_FILESYSTEM_EDITOR` for exceptions).

| ROLE_ASSET_EDITOR
| Permissions only to update asset records from nodes.

| ROLE_FILESYSTEM_EDITOR
| Permissions only to view and update file configuration data via the REST API. +
Note that even `ROLE_ADMIN` cannot view or edit configurations unless they also have the `ROLE_FILESYSTEM_EDITOR` role.
Also, for a user with `ROLE_FILESYSTEM_EDITOR` to use the UI, they will also need the `ROLE_USER` or similar role.

| ROLE_DASHBOARD
| Permissions only to view the dashboard.

| ROLE_DELEGATE

> | Allow actions (such as acknowledging an alarm) to be performed on behalf of another user.

Reviewer: ```suggestion
| Permissions to perform actions (such as acknowledging an alarm) on behalf of another user.
```

| ROLE_DEVICE_CONFIG_BACKUP
| Permissions to view and trigger device configuration backups.

| ROLE_FLOW_MANAGER
| Permissions to edit flow classifications.

| ROLE_JMX

> | Allows retrieval of JMX metrics but does not allow executing MBeans of the {page-component-title} JVM, even if they just return simple values.

Reviewer: ```suggestion
| Permissions to retrieve JMX metrics but does not not to execute MBeans of the {page-component-title} JVM, even if they just return simple values.
```

ifndef::opennms-prime[]

| ROLE_MINION
| Minimum required permissions for a Minion to operate.
endif::opennms-prime[]

| ROLE_MOBILE
| Permissions to use OpenNMS COMPASS mobile application to acknowledge alarms and notifications via the REST API.

| ROLE_PROVISION
| Permissions to use the provisioning system and configure SNMP in {page-component-title} to access management information from devices.

| ROLE_READONLY ^*^
| Permissions only to read information in the web UI; user cannot change alarm states or notifications.

| ROLE_REPORT_DESIGNER
| Permissions to manage reports in the web UI and REST API.

| ROLE_REST
| Permissions to interact with the entire {page-component-title} REST API.

| ROLE_RTC ^*^
| Permissions to exchange information with the {page-component-title} Real-Time Console for availability calculations.

| ROLE_USER ^*^
| Default permissions for a new user to interact with the web UI: can escalate and acknowledge alarms and notifications.
|===

== Assign security roles

Follow the steps below to assign security roles to user accounts:

> . Click the *Gear* symbol at the top-right of the screen.

Reviewer: We can leave this for now, until we decide to formally go back to not bolding symbols. :-)

> | Allow user to use OpenNMS COMPASS mobile application to acknowledge alarms and notifications via the REST API.

Reviewer: ```suggestion
| Permissions to use OpenNMS COMPASS mobile application to acknowledge alarms and notifications via the REST API.
```

| ROLE_PROVISION
| Permissions to use the provisioning system and configure SNMP in {page-component-title} to access management information from devices.

> | ROLE_READONLY \*

Reviewer: ```suggestion
| ROLE_READONLY ^*^
```

> | User limited to reading information in the web UI; unable to change alarm states or notifications.

Reviewer: ```suggestion
| Permissions only to read information in the web UI; user cannot change alarm states or notifications.
```

| ROLE_REPORT_DESIGNER
| Permissions to manage reports in the web UI and REST API.

| ROLE_REST
| Permissions to interact with the entire {page-component-title} REST API.

> | ROLE_RTC \*

Reviewer: ```suggestion
| ROLE_RTC ^*^
```

| Permissions to exchange information with the {page-component-title} Real-Time Console for availability calculations.

> | ROLE_USER \*

Reviewer: ```suggestion
| ROLE_USER ^*^
```

| Default permissions for a new user to interact with the web UI: can escalate and acknowledge alarms and notifications.
|===

== Assign security roles

Follow the steps below to assign security roles to user accounts:

. Click the *Gear* symbol at the top-right of the screen.
. Under Configure OpenNMS, click menu:Configure Users, Groups and On-Call Roles[Configure Users].
. Find the user to whom you want to assign a security role and click *Modify* beside their name.

> . Select the role to be added to the user account from the *Available Roles* list, and click *Add*.

Reviewer: ```suggestion
. Select the role to add to the user account from the *Available Roles* list, and click *Add*.
```
